test/jdk/java/util/Locale/bcp47/TExtensionTests.java

```diff
@@ -0,0 +1,193 @@
+    public void test_GetDisplayName(Locale locale, Locale inLocale, String expected) {
+        assertEquals(locale.getDisplayName(inLocale), expected);
+    }
+
+    @Test
+    public void test_FieldOrder() {
+        // order of the fields does NOT matter
+        var l = Locale.forLanguageTag("ja-u-nu-jpan-t-en-Latn-d0-fwidth-s0-ascii-x-pri-vate");
+        assertEquals(l, L4);
+    }
+
+    @Test
+    public void test_SubtagOrderInField() {
+        // order of the subtags in a field DOES matter
+        var l = Locale.forLanguageTag("und-Cyrl-t-und-latn-m0-2007-ungegn");
+        assertNotSame(l, L1);
+    }
+
+    @Test
+    public void test_FieldDuplicatesForLanguageTag() {
+        // Locale.forLanguageTag() should ignore the t extension
+        assertEquals(Locale.forLanguageTag("en-t-" + DUPLICATE), Locale.ENGLISH);
+    }
+
+    @Test
+    public void test_FieldDuplicatesSetExtension() {
+        // Locale.Builder.setExtension() should throw IllformedLocaleException
+        try {
+            LB.clear().setLocale(Locale.ENGLISH).setExtension('t', DUPLICATE).build();
+            throw new RuntimeException("Duplicated fields should throw an exception.");
+        } catch (IllformedLocaleException ile) {
+            assertEquals(ile.getErrorIndex(), DUPLICATE.indexOf(DUPLICATE_FIELD));
+            // success
+            System.out.println("IllformedLocaleException thrown correctly: " + ile.getMessage());
+        }
+    }
+
+    @Test
+    public void test_InvalidSourceLangForLanguageTag() {
+        // Locale.forLanguageTag() should ignore the t extension
+        assertEquals(Locale.forLanguageTag("en-t-" + INVALID_SOURCE), Locale.ENGLISH);
+    }
+
+    @Test
+    public void test_InvalidSourceLangSetExtension() {
+        // Locale.Builder.setExtension() should throw IllformedLocaleException
+        try {
+            LB.clear().setLocale(Locale.ENGLISH).setExtension('t', INVALID_SOURCE).build();
+            throw new RuntimeException("Invalid source language tag should throw an exception");
+        } catch (IllformedLocaleException ile) {
+            // success
+            System.out.println("IllformedLocaleException thrown correctly: " + ile.getMessage());
+        }
+    }
+
+    @Test
+    public void test_InvalidFieldsForLanguageTag() {
+        // Locale.forLanguageTag() should ignore the t extension
+        assertEquals(Locale.forLanguageTag("en-t-" + INVALID_FIELDS), Locale.ENGLISH);
+    }
+
+    @Test
+    public void test_InvalidFieldsSetExtension() {
+        // Locale.Builder.setExtension() should throw IllformedLocaleException
+        try {
+            LB.clear().setLocale(Locale.ENGLISH).setExtension('t', INVALID_FIELDS).build();
+            throw new RuntimeException("Invalid fields should throw an exception");
+        } catch (IllformedLocaleException ile) {
+            // success
+            System.out.println("IllformedLocaleException thrown correctly: " + ile.getMessage());
```

Comment:
Could use assertThrows, but ok if you want to keep it this way.

Reply:
Thanks, Joe. Replaced them with `assertThrows`.